File: Thread-Benchmark/1-NewThread/Case-2/MainActivity.java
package com.test.anr.backgroundThread;

import android.app.Activity;
import android.os.Bundle;
import java.io.FileWriter;
import java.io.IOException;

public class MainActivity extends Activity {

    private FileWriter fileWriter;
    Object object = new Object();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        fileWriter = null;
        object = null;
        IOBlockingCall("fromUIThread");
        case02_LambdaRunnable();
    }
    // case02: new Thread(() -> ...).start()
    private void case02_LambdaRunnable() {
        new Thread(() -> IOBlockingCall("case02-lambda-runnable")).start();
    }

    private void IOBlockingCall(String tag) {
        try {
            object.toString();
            fileWriter = new FileWriter(tag + ".txt");
            fileWriter.write("IOBlockingCall: " + tag + "\n");
            fileWriter.flush();
        } catch (IOException e) {
        }
    }
}

File: Thread-Benchmark/6-coroutines/Case-10/MainActivity.kt
package com.test.anr.backgroundThread

import android.app.Activity
import android.os.Bundle
import java.io.FileWriter
import java.io.IOException
import kotlinx.coroutines.*

class MainActivity : Activity() {

    private var fw: FileWriter? = null
    private val supervisorScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        try {
            fw = FileWriter("s.txt")
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
        runCase()
        runCase()
    }

    private fun runCase() {
        supervisorScope.launch {
            IOBlockingCall("case10-supervisor-scope-io")
        }
    }

    private fun IOBlockingCall(tag: String) {
        try {
            fw?.write("IOBlockingCall: $tag\n")
            fw?.flush()
        } catch (e: IOException) {
        }
    }
}
